src/audio/audio.cpp

```diff
@@ -486,16 +498,30 @@ void Audio::doCapture()
     }
 #endif
 
-    for (size_t i = 0; i < AUDIO_FRAME_SAMPLE_COUNT * AUDIO_CHANNELS; ++i)
-        buf[i] *= inGain;
+    for (quint32 i = 0; i < AUDIO_FRAME_SAMPLE_COUNT * AUDIO_CHANNELS; ++i)
+    {
+        int16_t ampPCM = static_cast<int16_t>(qRound(buf[i] * gain));
+        buf[i] = static_cast<int16_t>(ampPCM);
+    }
 
     emit frameAvailable(buf, AUDIO_FRAME_SAMPLE_COUNT, AUDIO_CHANNELS, AUDIO_SAMPLE_RATE);
 }
 
 /**
+@internal
+*/
+void Audio::privateSetGain(qreal percentage)
+{
+    percentage = std::max(0.0, std::min(percentage, 1.0));
+    gainPercentage = percentage;
+
+    gain = (percentage > 0.0) ? 10.0 * qAbs(log10(percentage * 100.0)) : 0.0;
```

Comment:
Removing the abs is no better since now gain can go negative (causing channel inversion) with the same effect (increasing negative gain). If this new gain curve is to be used, the log value has to be clipped at some lower bound (0 for example, such that it doesn't go negative).